ApplicationLibCode/Commands/CompletionCommands/RicImportValveTemplatesFeature.cpp

```diff
@@ -75,7 +75,25 @@ void RicImportValveTemplatesFeature::onActionTriggered( bool isChecked )
         }
 
         // There can be multiple items of the same template, make sure we have unique templates
-        std::sort( aicdTemplates.begin(), aicdTemplates.end() );
+        std::sort( aicdTemplates.begin(),
+                   aicdTemplates.end(),
+                   []( RiaOpmParserTools::AicdTemplateValues& templateA, RiaOpmParserTools::AicdTemplateValues& templateB )
+                   {
+                       int idA = 0;
+                       int idB = 0;
+
+                       if ( templateA.contains( RiaOpmParserTools::aicdTemplateId() ) )
+                       {
+                           idA = templateA.at( RiaOpmParserTools::aicdTemplateId() );
+                       }
```

Comment:
Since `AicdTemplateValues` [is a `map<>`](https://github.com/OPM/ResInsight/blob/9b797fdd36d6a765a8bdd555c9056df965efcca2/ApplicationLibCode/ProjectDataModel/RiaOpmParserTools.h#L39), you're effectively looking up the element twice when running `contains()` followed by `at()`.  It **probably** doesn't matter here, but in case it does it's faster to use `find()` and comparing the iterator to `end()` instead.

Reply:
Thanks for the feedback, I have updated based on your comments. It is always good practice to write code as efficient as possible, as we often do cut/paste of code.